File: Caracteristique.cs
namespace exercice_trombone
{
    /// <summary>
    /// Caractéristique que peut posséder une usine
    /// </summary>
    internal class Caracteristique
    {

        /// <summary>
        /// La valeur de la caractéristique.
        /// </summary>
        public int Valeur { get; set; }

        /// <summary>
        /// Le nom de la caractéristique.
        /// </summary>
        public string Nom { get; private set; }

        /// <summary>
        /// Initialise une nouvelle instance de la classe Caracteristique.
        /// </summary>
        /// <param name="nom">Le nom de la nouvelle caractérique</param>
        /// <param name="valeur">La valeur de la nouvelle caractérique</param>
        public Caracteristique(string nom, int valeur)
        {
            Nom = nom;
            Valeur = valeur;
        }


    }
}

File: src/edu/cegepvicto/usine/Caracteristique.java
package edu.cegepvicto.usine;

/**
 * Gère la valeur d'une des caractéristiques de l'usine.
 */
public class Caracteristique {

    /**
     * Valeur de la caractéristique
     */
    private int valeur;

    /**
     * Nom de la caractéristique
     */
    private String nom;

    /**
     * Contruit une nouvelle caractéristique.
     * @param nom le nom de la caractéristique.
     * @param valeur la valeur associée à la caractéristique.
     */
    public Caracteristique(String nom, int valeur) {
        this.nom = nom;
        this.valeur = valeur;
    }

    /**
     * Accesseur de la valeur de la caractéristique.
     * @return la valeur de la caractéristique
     */
    public int getValeur() {
        return valeur;
    }

    /**
     * Mutateur de la valeur de la caractéristique.
     * @param valeur la nouvelle valeur de la caractéristique.
     */
    public void setValeur(int valeur) {
        this.valeur = valeur;
    }

    /**
     * Accesseur du nom de la caractéristique.
     * @return le nom de la caractéristique.
     */
    public String getNom() {
        return nom;
    }

}
